src/main/java/org/apache/cassandra/sidecar/config/yaml/S3ClientConfigurationImpl.java

```diff
@@ -51,26 +68,34 @@ public S3ClientConfigurationImpl()
         this(DEFAULT_THREAD_NAME_PREFIX,
              DEFAULT_S3_CLIENT_CONCURRENCY,
              DEFAULT_THREAD_KEEP_ALIVE_SECONDS,
+             DEFAULT_RANGE_GET_OBJECT_BYTES_SIZE,
+             DEFAULT_API_CALL_TIMEOUT_MILLIS,
              new S3ProxyConfigurationImpl());
     }
 
     public S3ClientConfigurationImpl(String threadNamePrefix,
                                      int concurrency,
                                      long threadKeepAliveSeconds,
+                                     int rangeGetObjectBytesSize,
+                                     long apiCallTimeoutMillis,
                                      S3ProxyConfiguration proxyConfig)
     {
+        Preconditions.checkArgument(apiCallTimeoutMillis > TimeUnit.SECONDS.toMillis(10),
```

Comment:
Why is 10 seconds the minimum? Just an AWS client limitation or ?

Reply:
AWS client do not enforce any timeout for api calls by default. The 10 seconds is an arbitrary value that should prevent unexpected early exit in most case. The default api call timeout set as part of the patch is 60 seconds.